### PR TITLE
Upgraded jackson to 2.10.0, updated config of the ObjectMapper used by the RestTemplate

### PR DESCRIPTION
**Pull request by YetAnotherMatt**

### CODE REVIEW

**Review by dwybourn — DISMISSED** [review]

File: common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java [hunk 1]
```diff
@@ -77,7 +80,9 @@ public ObjectMapper objectMapper() {
         return new ObjectMapper()
                 .findAndRegisterModules()
                 .registerModule(javaTimeModule)
-                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+                .configure(MapperFeature.DEFAULT_VIEW_INCLUSION, false)
+                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
```
Comment (mikeSmith08) [inline common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java]: According to the Spring documentation, these values are the defaults, so any reason why you needed to add them?

File: common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java [hunk 2]
```diff
@@ -77,7 +80,9 @@ public ObjectMapper objectMapper() {
         return new ObjectMapper()
                 .findAndRegisterModules()
                 .registerModule(javaTimeModule)
-                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+                .configure(MapperFeature.DEFAULT_VIEW_INCLUSION, false)
```
Comment (mikeSmith08) [inline common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java:89]: Did you know that `configure` taking in a `MapperFeature` is deprecated? Can we look for an alternative?

File: common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java (hunk 1 above)
Comment (YetAnotherMatt) [reply common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java]: This isn't spring managed. I've updated the comment to make that clearer.

File: common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java (hunk 2 above)
Comment (YetAnotherMatt) [reply common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java:89]: Updated to the new format

Comment (mikeSmith08) [reply common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java:89]: Ok, but shouldn't we now remove `.configure(MapperFeature.DEFAULT_VIEW_INCLUSION, false)`?

**Review by dwybourn — APPROVED** [review]

File: common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java (hunk 2 above)
Comment (YetAnotherMatt) [reply common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/RequestContextConfiguration.java:89]: You might notice that we're now configuring that on the JsonMapper, not the ObjectMapper

**Review by mikeSmith08 — APPROVED** [review]

**Pull request merged**